pkg/machine/shim/host.go

```diff
@@ -485,6 +498,30 @@ func Start(mc *vmconfigs.MachineConfig, mp vmconfigs.VMProvider, opts machine.St
 		}
 	}
 
+	// Do not do anything with the system connection if its already
+	// the default system connection.
+	if !conn.Default {
+		// Prompt for system connection update
+		if updateSystemConn == nil {
+			response, err := promptUpdateSystemConn()
```

Comment:
What I was trying to tell you is that this part should still be in cmd/. Only the EditConnectionConfig() can be here. But I guess it doesn't matter to much either way so if you want to leave this as is fine with me.